### PR TITLE
[Task] Language is nullable

### LINKED ISSUE #603: Changing asset metadata in grid breaks version
- Update asset metadata in grid
Image
- open asset detail and try to open just created version
Image

`
[2024-12-02T10:47:35.213352+00:00] app.ERROR: TypeError: Pimcore\Bundle\StudioBackendBundle\Version\Schema\CustomMetadataVersion::__construct(): Argument #2 ($language) must be of type string, null given, called in /var/www/html/vendor/pimcore/studio-backend-bundle/src/Version/Hydrator/CustomMetadataVersionHydrator.php on line 43 and defined in /var/www/html/vendor/pimcore/studio-backend-bundle/src/Version/Schema/CustomMetadataVersion.php:32 Stack trace: #0 /var/www/html/vendor/pimcore/studio-backend-bundle/src/Version/Hydrator/CustomMetadataVersionHydrator.php(43): Pimcore\Bundle\StudioBackendBundle\Version\Schema\CustomMetadataVersion->__construct('author', NULL, 'input', 'WikimediaImages...') #1 /var/www/html/vendor/pimcore/studio-backend-bundle/src/Version/Hydrator/CustomMetadataVersionHydrator.php(36): Pimcore\Bundle\StudioBackendBundle\Version\Hydrator\CustomMetadataVersionHydrator->hydrateSingle(Array) #2 [internal function]: Pimcore\Bundle\StudioBackendBundle\Version\Hydrator\CustomMetadataVersionHydrator->Pimcore\Bundle\StudioBackendBundle\Version\Hydrator\{closure}(Array) #3 /var/www/html/vendor/pimcore/studio-backend-bundle/src/Version/Hydrator/CustomMetadataVersionHydrator.php(35): array_map(Object(Closure), Array) #4 /var/www/html/vendor/pimcore/studio-backend-bundle/src/Version/Hydrator/AssetVersionHydrator.php(69): Pimcore\Bundle\StudioBackendBundle\Version\Hydrator\CustomMetadataVersionHydrator->hydrate(Array) #5 /var/www/html/vendor/pimcore/studio-backend-bundle/src/Version/Hydrator/AssetVersionHydrator.php(57): Pimcore\Bundle\StudioBackendBundle\Version\Hydrator\AssetVersionHydrator->hydrateAsset(Object(Pimcore\Model\Asset\Image)) #6 /var/www/html/vendor/pimcore/studio-backend-bundle/src/Version/Service/VersionDetailService.php(128): Pimcore\Bundle\StudioBackendBundle\Version\Hydrator\AssetVersionHydrator->hydrate(Object(Pimcore\Model\Asset\Image)) #7 /var/www/html/vendor/pimcore/studio-backend-bundle/src/Version/Service/VersionDetailService.php(58): Pimcore\Bundle\StudioBackendBundle\Version\Service\VersionDetailService->hydrate(Object(Pimcore\Model\Asset\Image), 'Pimcore\\Model\\A...') #8 /var/www/html/vendor/pimcore/studio-backend-bundle/src/Version/Controller/GetController.php(74): Pimcore\Bundle\StudioBackendBundle\Version\Service\VersionDetailService->getVersionData(50, Object(Pimcore\Model\User)) #9 /var/www/html/vendor/symfony/http-kernel/HttpKernel.php(181): Pimcore\Bundle\StudioBackendBundle\Version\Controller\GetController->getVersions(50) #10 /var/www/html/vendor/symfony/http-kernel/HttpKernel.php(76): Symfony\Component\HttpKernel\HttpKernel->handleRaw(Object(Symfony\Component\HttpFoundation\Request), 1) #11 /var/www/html/vendor/symfony/http-kernel/Kernel.php(197): Symfony\Component\HttpKernel\HttpKernel->handle(Object(Symfony\Component\HttpFoundation\Request), 1, true) #12 /var/www/html/vendor/symfony/runtime/Runner/Symfony/HttpKernelRunner.php(35): Symfony\Component\HttpKernel\Kernel->handle(Object(Symfony\Component\HttpFoundation\Request)) #13 /var/www/html/vendor/autoload_runtime.php(29): Symfony\Component\Runtime\Runner\Symfony\HttpKernelRunner->run() #14 /var/www/html/public/index.php(20): require_once('/var/www/html/v...') #15 {main} [] []
 `

### PR DESCRIPTION
**Pull request by mattamon**
## Changes in this pull request
Resolves #603

### CODE REVIEW

File: src/Version/Schema/CustomMetadataVersion.php [hunk 1]
```diff
@@ -24,16 +24,16 @@
...
     type: 'object'
 )]
 final readonly class CustomMetadataVersion
 {
     public function __construct(
         #[Property(description: 'Name', type: 'string', example: 'custom_metadata')]
         private string $name,
-        #[Property(description: 'Language', type: 'string', example: 'en')]
-        private string $language,
+        #[Property(description: 'Language', type: 'string', example: 'en', nullable: true)]
```
Comment (martineiber) [inline src/Version/Schema/CustomMetadataVersion.php]: getLanguage should then be also nullable!

**Review by martineiber — CHANGES_REQUESTED** [review]

File: src/Version/Schema/CustomMetadataVersion.php (hunk 1 above)
Comment (mattamon) [reply src/Version/Schema/CustomMetadataVersion.php]: You are totally right! Somehow I always forget the getter! Thanks! good catch!

File: src/Version/Schema/CustomMetadataVersion.php [hunk 2]
```diff
@@ -24,15 +24,15 @@
  */
 #[Schema(
     title: 'CustomMetadataVersion',
-    required: ['name', 'type', 'data'],
+    required: ['name', 'language', 'type', 'data'],
```
Comment (martineiber) [inline src/Version/Schema/CustomMetadataVersion.php]: if nullable then it should not be required?

**Review by martineiber — CHANGES_REQUESTED** [review]

**Review by martineiber — APPROVED** [review]
LGTM

**Pull request merged**